### PR TITLE
chore: enable forceTypeAssert linter

### LINKED ISSUE #663: forcetypeassert

### PR DESCRIPTION
**Pull request by kehiy**
- Fixes #663

### CODE REVIEW

File: committee/committee.go [hunk 1]
```diff
@@ -90,8 +90,12 @@ func (c *committee) Update(lastRound int16, joined []*validator.Validator) {
 	}
 
 	sort.SliceStable(oldestFirst, func(i, j int) bool {
-		return oldestFirst[i].Value.(*validator.Validator).LastSortitionHeight() <
-			oldestFirst[j].Value.(*validator.Validator).LastSortitionHeight()
+		valI, valIok := oldestFirst[i].Value.(*validator.Validator)
```
Comment (b00f) [inline committee/committee.go:93]: Can't we use generic here to avoid this type conversion?

File: committee/committee.go [hunk 2]
```diff
@@ -204,8 +212,11 @@ func (c *committee) String() string {
 // iterate uses for easy iteration over validators in list.
 func (c *committee) iterate(consumer func(*validator.Validator) (stop bool)) {
 	for e := c.validatorList.Front(); e != nil; e = e.Next() {
-		if consumer(e.Value.(*validator.Validator)) {
-			return
+		val, ok := e.Value.(*validator.Validator)
```
Comment (b00f) [inline committee/committee.go:215]: Similar here

File: crypto/bls/private_key.go [hunk 3]
```diff
@@ -143,5 +143,9 @@ func (prv *PrivateKey) PublicKey() crypto.PublicKey {
 }
 
 func (prv *PrivateKey) EqualsTo(right crypto.PrivateKey) bool {
-	return prv.fr.Equal(&right.(*PrivateKey).fr)
+	prvKey, ok := right.(*PrivateKey)
```
Comment (b00f) [inline crypto/bls/private_key.go:146]: @amirvalhalla is looking for a way to avoid these casts, if it is possible.

**Review by b00f — COMMENTED** [review]
I think we can refactor some logic to avoid these casts

File: util/testsuite/testsuite.go [hunk 4]
```diff
@@ -291,7 +297,7 @@ func (ts *TestSuite) GenerateTestCertificate(blockHash hash.Hash) *block.Certifi
 	_, priv3 := ts.RandomBLSKeyPair()
 	_, priv4 := ts.RandomBLSKeyPair()
 
-	sigs := []*bls.Signature{
+	sigs := []*bls.Signature{ //nolint
```
Comment (b00f) [inline util/testsuite/testsuite.go:300]: Why?

Comment (kehiy) [reply util/testsuite/testsuite.go:300]: it makes tests more heavy and complex.

File: committee/committee.go (hunk 1 above)
Comment (kehiy) [reply committee/committee.go:93]: I don't have idea now, I will search and try to make it done

File: util/testsuite/testsuite.go (hunk 4 above)
Comment (b00f) [reply util/testsuite/testsuite.go:300]: Hopefully we can use generics here. @amirvalhalla any update?

Comment (kehiy) [reply util/testsuite/testsuite.go:300]: okay, I will wait for updates from him.
also, I can start work on this if you wish.

Comment (b00f) [reply util/testsuite/testsuite.go:300]: Yes, You may try as well

Comment (amirvalhalla) [reply util/testsuite/testsuite.go:300]: @b00f 
@kehiy 

I will work on make generic these methods until end of this weekend , please be patient :)

Comment (amirvalhalla) [reply util/testsuite/testsuite.go:300]: @b00f 
@kehiy 

Hi guys
Unfortunately, right now we can't use generics for `PrivateKey`, `PublicKey`,`Signature` etc... the reason is if we want to use generics we should change the structure of a lot of functions and interfaces to support generics.

e.g. in the `bls` package we implemented `PrivateKey` interface methods so if we want to use generics for `EqualsTo`  we should convert `PrivateKey` struct to `PrivateKey[T any]` this change makes some conflicts for other methods  like `PrivateKeyFromString` method, another reason is `Crypto.PrivateKey` interface is not generic so if we also change it to generic like `Crypto.PrivateKey[T any]` another issue is we can't cast it to another type.(if want to do that we should use embedded struct).

if generics are necessary which I think we should consider it, we should create another issue and discuss about what methods or packages should be generic then refactor our codes.

File: committee/committee.go (hunk 1 above)
Comment (kehiy) [reply committee/committee.go:93]: fixed in :
- #667

File: committee/committee.go (hunk 2 above)
Comment (kehiy) [reply committee/committee.go:215]: fixed in:
- #667

**Comment by kehiy** [comment]
@b00f @amirvalhalla I'm going to close this issue for Conflicting files and after trying to fix some type assertions I will open a new PR for this linter.

**Pull request closed without merging**